### PR TITLE
added favorites links to cards

### LINKED ISSUE #16: add favorites link to credit card cards

### PR DESCRIPTION
**Pull request by JoseMDO**

### CODE REVIEW

File: app/controllers/favorites_controller.rb [hunk 1]
```diff
@@ -4,10 +4,20 @@ class FavoritesController < ApplicationController
   # GET /favorites or /favorites.json
   def index
     @favorites = Favorite.all
+    @user_transactions = UserTransaction.all
+    @credit_cards = []
+    @favorites.each do |favorite|
+      @credit_cards.append(CreditCard.find_by(id: favorite.credit_card_id))
```
Comment (marywebby) [inline app/controllers/favorites_controller.rb:10]: good logic here to add the CreditCard by ID and append it directly to the creditcards array to use later, very handy

**Review by Dylanm22 — APPROVED** [review]

File: app/views/credit_cards/_credit_card.html.erb [hunk 2]
```diff
@@ -1,6 +1,14 @@
...
   <div class="card-body">
-    <h5 class="card-title"><%= credit_card.name %></h5>
+    <div class="d-flex">
+      <h5 class="card-title"><%= credit_card.name %></h5>
+      <% if favorite = Favorite.find_by(user_id: current_user, credit_card_id: credit_card.id)  %>
+        <%= link_to 'Unfavorite', favorite, data: { turbo_method: :delete }, class: "btn btn-danger" %>
+      <% else %>
+        <%= render "favorites/form", favorite: Favorite.new, credit_card_id: credit_card.id %>
+      <% end %>
+      <%#= render "favorites/favorite_unfavorite", favorite: Favorite.find_by(user_id: current_user, credit_card_id: credit_card.id) %>
```
Comment (marywebby) [inline app/views/credit_cards/_credit_card.html.erb:10]: if this logic isnt needed anymore i would recommend deleting it! helps tidy up the code

File: app/views/favorites/_form.html.erb [hunk 3]
```diff
@@ -12,16 +12,14 @@
   <% end %>
 
   <div>
-    <%= form.label :credit_card_id, style: "display: block" %>
-    <%= form.text_field :credit_card_id %>
+    <%= form.hidden_field :user_id, value: current_user.id %>
```
Comment (marywebby) [inline app/views/favorites/_form.html.erb:15]: good job making these hidden!

File: app/views/favorites/index.html.erb [hunk 4]
```diff
@@ -1,14 +1,23 @@
...
-  <% @favorites.each do |favorite| %>
-    <%= render favorite %>
-    <p>
-      <%= link_to "Show this favorite", favorite %>
-    </p>
-  <% end %>
-</div>
+<div class="container">
+  <h1>Favorites</h1>
+<!--  <div id="user_transactions">-->
```
Comment (marywebby) [inline app/views/favorites/index.html.erb:5]: again here with getting rid of unnecessary commented out code

**Review by marywebby — APPROVED** [review]
Everything's looking good! nice progress!

File: app/views/layouts/application.html.erb [hunk 5]
```diff
@@ -28,7 +27,6 @@
       <% end %>
 
       <%= yield %>
-    </div>
```
Comment (Dylanm22) [inline app/views/layouts/application.html.erb:31]: Unneeded spaces can be taken out to make code look even cleaner!

File: app/views/home/index.html.erb [hunk 6]
```diff
@@ -75,3 +75,5 @@
     </div>
   </div>
 </div>
+
```
Comment (Dylanm22) [inline app/views/home/index.html.erb:78]: Unneeded spaces

File: app/views/favorites/index.html.erb (hunk 4 above)
Comment (Dylanm22) [inline app/views/favorites/index.html.erb:5]: If the commented out code isn’t needed it can be taken out

**Pull request merged**